### ISSUE TITLE
[Warlock] Infernal Minion

### ISSUE THREAD

**Issue by heaventhegod17** [dalaranwow/dalaran-wow#5683]
[//]: # (INFORMATION ABOUT TAGS)
[//]: # (The black and yellow tag is unique. There can only be one label assigned to them - one black and one yellow)
[//]: # (Priority - ... tags are assigned by team members)
[//]: # (Your issue will be reviewed by testers and the developers will fix it in the future.)
[//]: # (Only once the issue got the label [Fixed - On Live Server] you can find the fix on your Gameserver.)
[//]: # (You can increase the relevance of your issue by adding a thumb-up emote http://i.imgur.com/RFouVVi.png)

[//]: # (***************************************************************)
[//]: # (** !!!!!!!!!!!!!DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED!!!!!!!!!!!! **)
[//]: # (***************************************************************)

**Current behaviour**: Tell us what happens in detail (Where, When, What, Why, Who)
So i was LVL 37 Warlock, went ahead to The Barrens to do the mission where i can choose either to get a fel guard or an infernal as a minion. I choose the Infernal, go all the way to Desolace where i kill with ease an Infernal and capture him using the crystal i was given, after that i am sent all the way to Dustwallow Marsh where i speak with an NPC and spawns a Fel Guard that i am supposed to kill. I fail into killing the fail guard so the mission is set as failed. I abandon to try and do it again later but now it seems i no longer can get that mission.
**Expected behaviour**: Tell us what should happen instead
I suppose i should be able to get the mission again.
**Steps to reproduce the problem**:

1.  get the warlock missions from stormwind (at the trainers) that send you all the way to the barrens
2. go to ratchet
3. talk with the npc there that will let you choose between 2 of other npcs from inside that tower, pick the one on the left with the infernal
4. go to desolace and capture an infernal as said in mission
5. go back to ratchet, give the mission in and get sent to dustwallow marsh
6. go there, talk with the lady, fail to kill the fel guard
7. go back to stormwind and try to get the mission back

**Include proofs for this behaviour**

You can research this on several pages like http://wowhead.com or http://wowwiki.wikia.com or http://www.youtube.com/
Issues without proofs will be closed and tag On Hold will be added until the proof is provided.

**Include the ID for the game objects, npcs (creatures,pets,minions), spells, items, quests, instances, zones, achievements, skills**

You can research this on several pages like https://wowgaming.altervista.org/aowow/ or https://wowhead.com/

**Include Screenshots from the issue if necessary**
